### ISSUE TITLE
Missing and mismatched keys in loading pre-trained checkpoint

### ISSUE THREAD

**Issue by chenqi-li** [Edoar-do/HuBERT-ECG#3]
Hi,

Thanks for the interesting work and quick responses! 

I am trying to use the pre-trained model, in particular, `hubert_ecg_small.pt` from Hugging face.

I have downloaded the checkpoint and tried to load it with the code snippet from Hugging face:

```
path = "./checkpoint/hubert_ecg_small.pt"
checkpoint = torch.load(path, map_location='cpu')
config = checkpoint['model_config']
hubert_ecg = HuBERTECG(config)
hubert_ecg.load_state_dict(checkpoint['model_state_dict'])
```

I get the error: 
```
RuntimeError: Error(s) in loading state_dict for HuBERTECG:
	Missing key(s) in state_dict: "encoder.pos_conv_embed.conv.weight_g", "encoder.pos_conv_embed.conv.weight_v". 
	Unexpected key(s) in state_dict: "encoder.pos_conv_embed.conv.parametrizations.weight.original0", "encoder.pos_conv_embed.conv.parametrizations.weight.original1". 
```

Why are there missing and unexpected keys? Is this the case on your end as well?

I tried to do the same with `hubert_ecg_base.pt` and got the same error.

Thanks in advance and happy holidays!

**Comment by Edoar-do** [comment]
Hi @chenqi-li 
everything works perfectly on my end. I downloaded the model from Hugging Face, copied your code and got `<All keys matched successfully>`. The error you're runnning into seems simply a name mismatch due to that `parametrizations`, something one can work around if necessary. However, here a list of things to check/do:
- Did you install the content of the requirements.txt file?
- Is you local/cloned repo aligned with this one?
- Try download the model again as I might've done some changes to the repos in the last 25 days
- alternatively, try using `AutoModel.from_pretrained`

**Comment by chenqi-li** [comment]
I have re-cloned the repo and re-downloaded the model from hugging face. It still gives me the same error. So I believe it should be a problem associated with my environment. I need additional packages from another repo, so I suspect some packages do not have the right version.

My machine currently do not have access to internet. Hopefully I will get internet back soon and update you with the cause.

Thanks for help!

**Comment by Edoar-do** [comment]
I think you're probably using a deprecated version of `torch.nn.utils.weight_norm`. The up-to-date version relies on `parametrization`. If you take a look at the official PyTorch guide on this matter you'll see that the problem is easier to solve than it seems: https://pytorch.org/docs/stable/generated/torch.nn.utils.weight_norm.html

**Comment by chenqi-li** [comment]
Indeed, I am using an older version of torch, so it uses `torch.nn.utils.weight_norm` instead of `torch.nn.utils.parametrization.weight_norm`. After some investigation, it seems that `encoder.pos_conv_embed.conv.weight_g` and `encoder.pos_conv_embed.conv.weight_g` are the same as `encoder.pos_conv_embed.conv.parametrizations.weight.original0` and `encoder.pos_conv_embed.conv.parametrizations.weight.original0`. So I modified the keys of the checkpoint and all weights have been loaded.

Is there anyway I can check that the model has loaded properly? I am thinking maybe would it be possible for you provide a few pairs of sample input and output of the model? I think this would be quite useful for others too, especially works that uses your model for multi-modal tasks. Sometimes it is not possible to match your environment exactly and this check would be great to make sure that everything is working as expected.

Thanks!

**Comment by Edoar-do** [comment]
You can try to reproduce the results in our paper. To do so, you can easily use the material in the `reproducibility` folder and exploit the `test.sh` for example
